### ISSUE TITLE
Return raw, signed, transaction when broadcasting final tx proposal.

### ISSUE THREAD

**Issue by lazerdye** [bitpay/bitcore-wallet-service#308]
For my application, we would like to have a record of all signed transactions sent using the wallet service. In the current implementation, the final signed transaction is generated when broadcasted, and is only on the server. 
The raw transaction could be returned by the broadcast function, but just adding it would be a breaking api change. I suggest a new parameter, "includeRaw" which, if set to "true", will return a new structure:
{
  "txproposal": { ... original transaction proposal struction ... },
  "rawTx": "... raw transaction in hex ..."
}

**Comment by matiu** [comment]
Are you using bitcore-wallet-client?

**Comment by lazerdye** [comment]
Hi, yes I am using bitcore-wallet-client, is there a way to get the raw transaction using it?

**Comment by matiu** [comment]
given a tx proposal, you can use WalletUtils.buildTx.

See the code of API.prototype.broadcastTxProposal at `bitcore-wallet-client`


On Wed, Aug 12, 2015 at 10:53 AM, Terence Haddock <notifications@github.com>
wrote:

> Hi, yes I am using bitcore-wallet-client, is there a way to get the raw
> transaction using it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitpay/bitcore-wallet-service/issues/308#issuecomment-130313676>
> .
>



-- 
Matías Alejo Garcia
@ematiu
Roads? Where we're going, we don't need roads!

**Comment by lazerdye** [comment]
Yes, I saw that, the problem I ran into is that the transaction returned there is unsigned. The getBitcoreTx function in /lib/model/txproposal.js calls _addSignaturesToBitcoreTx which is a (relatively) complex function to add the signatures to the tx proposal. Of course, I could do that as well by copying the code, it seemed a bad idea at first glance.
Perhaps this signature code could be moved to WalletUtils, and used on the client and the server, but I did not see if that is feasible.
